Integrated Bootstrap in the User Interface...

MoeweX opened the pull request:
... and made some minor improvements. 

![bildschirmfoto 2014-03-06 um 10 51 04](https://f.cloud.github.com/assets/5738978/2344069/e322685c-a519-11e3-80d8-8afd41f0d676.png)
Functionality remains the same
![bildschirmfoto 2014-03-06 um 10 51 17](https://f.cloud.github.com/assets/5738978/2344067/df901036-a519-11e3-960a-32bca092f753.png)

MoeweX committed:
Integrated Bootstrap in the User Interface and made some improvements and code clean ups.

StephanEwen commented:
Looks very good, I like it :-)

Before we can merge this, we need to make sure the legal stuff is cleared.

Can you please go though all external libraries / css / javascript / etc that you added and find out the license under which it is distributed? Please post the list here so we get an overview and can double check. It is a bit tedious, but actually really important!

MoeweX commented:
Most parts are from bootstrap/jquery -> MIT license: https://github.com/twbs/bootstrap/blob/master/LICENSE

*timeline.js:  "Licensed under the Apache License, Version 2.0 (the "License")"
*jcanvas.min.js": MIT license
*jquery.tablesorter.js: "Dual licensed under the MIT and GPL licenses"
*font-awesome: MIT license

I think that should be all

StephanEwen commented:
What about the images for the icons?

On Thu, Mar 6, 2014 at 11:45 AM, JonathanH5 notifications@github.comwrote:

> Most parts are from bootstrap/jquery -> MIT license:
> https://github.com/twbs/bootstrap/blob/master/LICENSE
> 
> *timeline.js: "Licensed under the Apache License, Version 2.0 (the
> "License")"
> *jcanvas.min.js": MIT license
> *jquery.tablesorter.js: "Dual licensed under the MIT and GPL licenses"
> *font-awesome: MIT license
> 
> I think that should be all
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/534#issuecomment-36843801
> .

MoeweX commented:
that is font-awesome: http://fortawesome.github.io/Font-Awesome/

StephanEwen commented:
Okay, good.

I am a bit confused about the dual license MIT / GPL. What does that mean? Both licenses have very different implications?

rmetzger commented:
I think it is possible that software has different licenses at the same
time.
For example there are two versions of jackson-core, one with a `-asl` and
one with a `-lgpl` suffix.

On Thu, Mar 6, 2014 at 12:08 PM, Stephan Ewen notifications@github.comwrote:

> Okay, good.
> 
> I am a bit confused about the dual license MIT / GPL. What does that mean?
> Both licenses have very different implications?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/534#issuecomment-36845341
> .

## 

Robert Metzger, Kontakt: metzgerr@web.de, Mobil: 0171/7424461

rmetzger commented on stratosphere-runtime/resources/web-docs-infoserver/index.html:
```diff
@@ -1,84 +1,168 @@
+    <link rel="stylesheet" href="font-awesome/css/font-awesome.min.css">
+
+    <!-- Page Specific CSS -->
+    <link rel="stylesheet" href="http://cdn.oesmith.co.uk/morris-0.4.3.min.css">
```
is it possible to ship all required CSS and JS files with our code?
This requires the user to have a working internet connection.

rmetzger commented on stratosphere-runtime/resources/web-docs-infoserver/analyze.html:
```diff
@@ -1,88 +1,128 @@
+
+    <!-- Page Specific Plugins -->
+    <script src="http://cdnjs.cloudflare.com/ajax/libs/raphael/2.1.0/raphael-min.js"></script>
+    <script src="http://cdn.oesmith.co.uk/morris-0.4.3.min.js"></script>
```
external dependencies.

rmetzger commented on stratosphere-runtime/resources/web-docs-infoserver/js/jobmanagerFrontend.js:
```diff
@@ -35,7 +35,7 @@ $(init());
 	$.ajax({ url : "jobsInfo?get=updates&job="+jobId, type : "GET", cache: false, success : function(json) {
 
-		// Call init of no more jobs are running
+		// Call init of no more jobs are running˝
```
remove the `˝`

MoeweX commented:
I removed the external dependencies and deleted not used files. If we need them in the future we can add them again (http://startbootstrap.com/sb-admin) @rmetzger

rmetzger commented:
The interface is beautiful.

But it does not work with YARN (the link in the archive to the `analyze.html` is an absolute path.
(We had this bug already https://github.com/stratosphere/stratosphere/issues/325).

fhueske commented:
:+1: really like it as well. Nicely done!

rmetzger commented:
When running a job the sums in the last row do not make sense. They are not the sums of the rows above.

StephanEwen commented:
I agree with @rmetzger 
If the sums are off, we will get questions about that. Can you have a look at that.

@markus-h Can you also review the pull request?

MoeweX commented:
Can you please specify in which case the sums are off or include a screenshot? I made some test and I can't see any differences
![kmeansiterative](https://f.cloud.github.com/assets/5738978/2408471/b7c7d8b2-aa9c-11e3-9e3a-5be99c2a5863.png)

MoeweX commented:
Changes of my last commit (colored collums):
![changes](https://f.cloud.github.com/assets/5738978/2408881/e95e80e0-aaa3-11e3-992d-9d04567e6d54.png)
I put it in it's own commit, because it is not only a bug fix, it is an improvement.

rmetzger commented:
The YARN bug still exists. (have a look at the URL in the screenshot)

And 16*3 =! 32 (Starting column)

Can you add some space below the "cancel" button ?

![wrong](https://f.cloud.github.com/assets/89049/2428444/bfa69f66-ac48-11e3-94c6-645efe9e59d2.png)

And:

![wrong1](https://f.cloud.github.com/assets/89049/2428464/3f74e0cc-ac49-11e3-96b5-cf794b94401d.png)

rmetzger commented:
I think it is okay this way.

MoeweX commented:
It is not, sorry for any inconvenience, I found a bug and I am working on it

MoeweX commented:
@markus-h  I think I fixed it, the old code is still inside the update function, I am going to delete it when you say my version works fine for you :).

I recognized that the Job-History only contains the last five jobs, is this on purpose?

rmetzger commented:
Hi,
you can configure the number of jobs stored in the history.

MoeweX commented:
Ok, good. So everything is fine in that regard.

MoeweX commented:
I integrated the changes from @qmlmoon in his commits f050b708268b781fd86cadd96aeaf6469ebec7e2 and 7aacbf3994cca859d4097ebea117934899a7b684 because it could not be done automatically.

In addition to that I created a placeholder message in the running jobs table
![bildschirmfoto 2014-03-25 um 14 33 50](https://f.cloud.github.com/assets/5738978/2512063/7a7411b2-b422-11e3-8bf0-64db370a1edd.png)

rmetzger commented:
Have you fixed the YARN bug?

MoeweX commented:
Of course!

StephanEwen commented:
Thanks!

Merged in cc8e16ebd53500c146dd9826436f983d3f243151

MoeweX commented:
Fine :), but @StephanEwen, why does it say: 
![bildschirmfoto 2014-03-26 um 01 07 11](https://f.cloud.github.com/assets/5738978/2519837/9a9bafd8-b47a-11e3-958f-001718813dda.png)

StephanEwen commented:
I rebased the commit to the head of the master branch. We do that usually to keep a straight and clean commit history. Rebasing the commit changes the hash, because the hash is based on content and preceding commits.

With a changed hash, git does not recognize the branch as merged. But your changes are in, don't worry ;-)

StephanEwen commented:
BTW: where do you see this message? Are you using some client?

MoeweX commented:
I see this message on the website ... . Ok, I rebased my changes to the last version I had in the afternoon. So I guess something was changed later on.
